src/com/android/calendar/icalendar/VEvent.java

```diff
@@ -0,0 +1,188 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.calendar.icalendar;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.UUID;
+
+/**
+ * Models the Event/VEvent component of the iCalendar format
+ */
+public class VEvent {
+
+    // Valid property identifiers for an event component
+    // TODO: only a partial list of attributes has been implemented, implement the rest
+    public static String CLASS = "CLASS";
+    public static String CREATED = "CREATED";
+    public static String LOCATION = "LOCATION";
+    public static String ORGANIZER = "ORGANIZER";
+    public static String PRIORITY = "PRIORITY";
+    public static String SEQ = "SEQ";
+    public static String STATUS = "STATUS";
+    public static String UID = "UID";
+    public static String URL = "URL";
+    public static String DTSTART = "DTSTART";
+    public static String DTEND = "DTEND";
+    public static String DURATION = "DURATION";
+    public static String DTSTAMP = "DTSTAMP";
+    public static String SUMMARY = "SUMMARY";
+    public static String DESCRIPTION = "DESCRIPTION";
+    public static String ATTENDEE = "ATTENDEE";
+    public static String CATEGORIES = "CATEGORIES";
+
+    // Stores the -arity of the attributes that this component can have
+    private static HashMap<String, Integer> sPropertyList = new HashMap<String, Integer>();
+
+    // Initialize the approved list of mProperties for a calendar event
+    static {
+        sPropertyList.put(CLASS,1);
+        sPropertyList.put(CREATED,1);
+        sPropertyList.put(LOCATION,1);
+        sPropertyList.put(ORGANIZER,1);
+        sPropertyList.put(PRIORITY,1);
+        sPropertyList.put(SEQ,1);
+        sPropertyList.put(STATUS,1);
+        sPropertyList.put(UID,1);
+        sPropertyList.put(URL,1);
+        sPropertyList.put(DTSTART,1);
+        sPropertyList.put(DTEND,1);
+        sPropertyList.put(DURATION, 1);
+        sPropertyList.put(DTSTAMP,1);
+        sPropertyList.put(SUMMARY,1);
+        sPropertyList.put(DESCRIPTION,1);
+
+        sPropertyList.put(ATTENDEE, Integer.MAX_VALUE);
+        sPropertyList.put(CATEGORIES, Integer.MAX_VALUE);
+        sPropertyList.put(CATEGORIES, Integer.MAX_VALUE);
+    }
+
+    // Stores attributes and their corresponding values belonging to the Event component
+    public HashMap<String, String> mProperties;
+
+    public LinkedList<Attendee> mAttendees;
+    public Organizer mOrganizer;
+
+    /**
+     * Constructor
+     */
+    public VEvent() {
+        mProperties = new HashMap<String, String>();
+        mAttendees = new LinkedList<Attendee>();
+
+        // Generate and add a unique identifier to this event - iCal requisite
+        addProperty(UID , UUID.randomUUID().toString() + "@android.local");
```

Comment:
Thinking of it, maybe we should change this to ws.xsoh.etar as well, as it only needs to be unique, afaik